pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -179,7 +179,13 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(proxy *model.Proxy,
 	}
 	networkView := model.GetNetworkView(proxy)
 
-	for _, service := range push.Services(proxy) {
+	var services []*model.Service
```

Comment:
Can we push this logic to` push.Services` function? Similar to how it handles Sidecar [here](https://github.com/istio/istio/blob/master/pilot/pkg/model/push_context.go#L519), it can also handle gateways based on new feature flag. That way cluster.go is unchanged.

Reply:
`push.Services` is invoked in sni-dnat case as well and in sni-dnat case, we want to create all the clusters and therefore push.Services is kept as it is for `buildOutboundSniDnatClusters`